### PR TITLE
Add user tolerance to Serial SVD

### PR DESCRIPTION
**Pull request by wjhorne**
Hi,

I have have had issues with stalling when attempting to use Serial SVD. The basic behavior is that the superdiagonal entries oscillate around at really small numbers, yet they are still above the threshold checked in bidiSVD to zero them out, resulting in an infinite loop. 

This PR introduces a user specified tolerance that can optionally be used to zero out tiny superdiagonal entries in bidiSVD. This change enabled the routine to work without issue for my problems. I ultimately used 2.0 * machine_eps as the tolerance rather than the value used by master.

I am not certain if this is the best strategy to deal with the issue and welcome other ideas on how to improve it.

### CODE REVIEW

**Comment by lucbv** [comment]
Hi @wjhorne Kokkos Kernels only accepts pull request against the develop branch, master is our release branch where you will see that we only perform updates when a new release comes out.
Can you rebase your branch against develop and push again with" `git push origin +my_local_branch_name`, thanks.
I will add reviewers and start reviewing you PR.

File: batched/dense/impl/KokkosBatched_SVD_Serial_Impl.hpp [hunk 1]
```diff
@@ -23,12 +23,14 @@
 namespace KokkosBatched {
 // Version which computes the full factorization
 template <typename AViewType, typename UViewType, typename VViewType,
-          typename SViewType, typename WViewType>
+          typename SViewType, typename WViewType, 
+          typename TolType>
 KOKKOS_INLINE_FUNCTION int SerialSVD::invoke(SVD_USV_Tag, const AViewType &A,
```
Comment (lucbv) [inline batched/dense/impl/KokkosBatched_SVD_Serial_Impl.hpp:28]: This function is part of our public API, the change you are making will break the code of users who already use this function. A better change would be to introduce the new interface like so:

```c++
template <typename AViewType, typename UViewType, typename VViewType,
          typename SViewType, typename WViewType, 
          typename TolType>
KOKKOS_INLINE_FUNCTION int SerialSVD::invoke(SVD_USV_Tag, const AViewType &A,
                                             const UViewType &U,
                                             const SViewType &sigma,
                                             const VViewType &Vt,
                                             const WViewType &work,
                                             const TolType tol) {
...
}

template <typename AViewType, typename UViewType, typename VViewType,
          typename SViewType, typename WViewType>
KOKKOS_INLINE_FUNCTION int SerialSVD::invoke(SVD_USV_Tag, const AViewType &A,
                                             const UViewType &U,
                                             const SViewType &sigma,
                                             const VViewType &Vt,
                                             const WViewType &work) {
   SerialSVD::invoke(SVD_USV_Tag, A, U, sigma, Vt, work, some_default_tol);
}
```

File: batched/dense/impl/KokkosBatched_SVD_Serial_Impl.hpp [hunk 2]
```diff
@@ -23,12 +23,14 @@
 namespace KokkosBatched {
 // Version which computes the full factorization
 template <typename AViewType, typename UViewType, typename VViewType,
-          typename SViewType, typename WViewType>
+          typename SViewType, typename WViewType, 
+          typename TolType>
```
Comment (lucbv) [inline batched/dense/impl/KokkosBatched_SVD_Serial_Impl.hpp:27]: Do we really need a new template paramter for TolType, would it make sense to use the something like:
`Kokkos::ArithTraits<typename AViewType::non_const_value_type>::mag_type`?

File: batched/dense/impl/KokkosBatched_SVD_Serial_Impl.hpp [hunk 3]
```diff
@@ -46,14 +48,16 @@ KOKKOS_INLINE_FUNCTION int SerialSVD::invoke(SVD_USV_Tag, const AViewType &A,
...
       A.extent(0), A.extent(1), A.data(), A.stride(0), A.stride(1), U.data(),
       U.stride(0), U.stride(1), Vt.data(), Vt.stride(0), Vt.stride(1),
-      sigma.data(), sigma.stride(0), work.data());
+      sigma.data(), sigma.stride(0), work.data(), tol);
 }
 
 // Version which computes only singular values
-template <typename AViewType, typename SViewType, typename WViewType>
+template <typename AViewType, typename SViewType, typename WViewType, 
+          typename TolType>
```
Comment (lucbv) [inline batched/dense/impl/KokkosBatched_SVD_Serial_Impl.hpp:56]: Same as above

**Review by lucbv — CHANGES_REQUESTED** [review]

**Comment by kokkos-devops-admin** [comment]
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           NO INSPECTION HAS BEEN PERFORMED ON THIS PULL REQUEST! - This PR must be inspected by setting label 'AT: PRE-TEST INSPECTED'.

**Pull request closed without merging**